serialization/murmur3/README.md

```diff
@@ -12,4 +10,184 @@ Run with:
+<h3>Key to Hash Code</h3>
+So, how do we find the partition for the key in the real world to avoid the pitfall that the key may not be numeric and keys may have echelon patterns?
+
+The solution is to use a hashing function, which will take any type of object and return a number. While the output number is not required to be unique (two keys can hash to the same value), we wish the range of output values to be evenly spread even if the range of input values is uneven.
+
+Java provides a “<code>hashCode()</code>” function for objects, which most IDEs will generate for you. This is not used. Partly because Hazelcast is used by non-Java code such as .NET and Python. And partly also because it doesn’t produce a particularly good spread of output values.
+
+Instead, we use the "<i>murmur hash 3</i>" algorithm, which is described <a href="https://en.wikipedia.org/wiki/MurmurHash#MurmurHash3">here</a> and in many other places.
+<h3>Murmur Hash 3</h3>
+Hazelcast's implementation of Murmur Hash 3 is <a href="https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/util/HashUtil.java">here for Java</a>, and with matching implementations for each of the client languages.
+
+The murmur algorithm is fairly complicated mathematics, with bit shifting and binary arithmetic. If you’re interested in going further, please follow the Wikipedia link to learn more. And if not, just regard it as a trusted function that gives the desired output.
+<h3>Why 271</h3>
+The final piece of this little puzzle is why 271.
+
+What is significant about 271 in this context is it is a prime number.
+
+We take the modulus of the hash output by the partition count to determine the partition. Although the hash function should give us a spread of output values for uneven input, using a prime number for the modulus further enhances this even data spread.
+
+For example, if the key is 1 the hash code might be 768969306. If we do 768969306 % 271, the modulus or remainder is 31. So integer key 1 belongs in partition 31. Partition 31 may not have that key, because it may not have been written, but we know the only and only partition to look in to confirm.
+<h3>Checkpoint</h3>
+So now we have an explanation.
+
+Code below will show that the keys 1, 2, 5, 9 and 50 are assigned to
+partitions 31, 5, 169, 42 and 105 respectively.
+
+In our 3-nodes cluster, the 271 partitions mean the nodes get 90 or 91 partitions each, the best we can spread them out. Recall that when we do this, we don't know what data will be inserted.
+
+At this point, it's not unreasonable that the node on the left of the diagram above owns partitions 5, 31 and 42, along with nearly ninety other partitions.
+
+In fact, if we only inserted records with keys 1, 2 and 9, we would find all our data on one node.
```

Comment:
You may mention if you want that this is an example of a "skewed" distribution.

Reply:
As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching